### PR TITLE
fix build message load/save

### PR DESCRIPTION
**Pull request by ansyral**

### CODE REVIEW

File: src/Microsoft.DocAsCode.Build.Engine/Incrementals/IncrementalUtility.cs [hunk 1]
```diff
@@ -98,7 +99,7 @@ public static void SaveBuildMessage(string fileName, BuildMessage bm)
             {
                 return;
             }
-            using (var writer = new StreamWriter(fileName))
+            using (var writer = new StreamWriter(fileName, false, Encoding.Unicode))
```
Comment (hellosnow) [inline src/Microsoft.DocAsCode.Build.Engine/Incrementals/IncrementalUtility.cs]: >Unicode [](start = 75, length = 7)

tests failed for [http://docfx-ci-0.cloudapp.net/viewLog.html?buildId=12423&buildTypeId=DocfxCiWithScripts_DocfxCiForDevBranch](http://docfx-ci-0.cloudapp.net/viewLog.html?buildId=12423&buildTypeId=DocfxCiWithScripts_DocfxCiForDevBranch)

File: src/Microsoft.DocAsCode.Build.Engine/Incrementals/PostProcessor/PostProcessInfo.cs [hunk 2]
```diff
@@ -81,7 +82,7 @@ internal void Save(string baseDir)
                 MessageInfoFile = IncrementalUtility.CreateRandomFileName(baseDir);
             }
             IncrementalUtility.SaveIntermediateFile(Path.Combine(baseDir, PostProcessOutputsFile), PostProcessOutputs);
-            using (var sw = new StreamWriter(Path.Combine(baseDir, MessageInfoFile)))
+            using (var sw = new StreamWriter(Path.Combine(baseDir, MessageInfoFile), false, Encoding.Unicode))
```
Comment (hellosnow) [inline src/Microsoft.DocAsCode.Build.Engine/Incrementals/PostProcessor/PostProcessInfo.cs]: >false [](start = 85, length = 5)

can remove the append parameter and use 2 param ctor?

File: src/Microsoft.DocAsCode.Build.Engine/Incrementals/IncrementalUtility.cs [hunk 3]
```diff
@@ -79,7 +80,7 @@ public static BuildMessage LoadBuildMessage(string file)
                 var content = JsonUtility.Deserialize<Dictionary<BuildPhase, string>>(reader);
                 foreach (var c in content)
                 {
-                    using (var sr = new StreamReader(Path.Combine(Path.GetDirectoryName(file), c.Value)))
+                    using (var sr = new StreamReader(Path.Combine(Path.GetDirectoryName(file), c.Value), Encoding.Unicode))
```
Comment (hellosnow) [inline src/Microsoft.DocAsCode.Build.Engine/Incrementals/IncrementalUtility.cs]: >Unicode [](start = 114, length = 7)

can remove the encoding? default will detectEncodingFromByteOrderMarks?

File: src/Microsoft.DocAsCode.Build.Engine/Incrementals/PostProcessor/PostProcessInfo.cs (hunk 2 above)
Comment (ansyral) [inline src/Microsoft.DocAsCode.Build.Engine/Incrementals/PostProcessor/PostProcessInfo.cs]: there's no 2 param ctor for StreamWriter

File: src/Microsoft.DocAsCode.Build.Engine/Incrementals/IncrementalUtility.cs (hunk 3 above)
Comment (ansyral) [inline src/Microsoft.DocAsCode.Build.Engine/Incrementals/IncrementalUtility.cs]: no, the pr is to handle unicode surrogates problem when splitting string

**Comment by hellosnow** [comment]
:shipit:

File: src/Microsoft.DocAsCode.Build.Engine/Incrementals/IncrementalUtility.cs [hunk 4]
```diff
@@ -7,13 +7,15 @@ namespace Microsoft.DocAsCode.Build.Engine.Incrementals
...
     using System.Linq;
+    using System.Text;
 
     using Microsoft.DocAsCode.Common;
     using Microsoft.DocAsCode.Plugins;
 
     public static class IncrementalUtility
     {
         private const int MaxRetry = 3;
+        private static Encoding UTF8 = new UTF8Encoding(false, false);
```
Comment (vwxyzh) [inline src/Microsoft.DocAsCode.Build.Engine/Incrementals/IncrementalUtility.cs]: readonly

File: src/Microsoft.DocAsCode.Build.Engine/Incrementals/PostProcessor/PostProcessInfo.cs [hunk 5]
```diff
@@ -14,6 +15,7 @@ namespace Microsoft.DocAsCode.Build.Engine.Incrementals
     public class PostProcessInfo
     {
         public const string FileName = "postprocess.info";
+        private static Encoding UTF8 = new UTF8Encoding(false, false);
```
Comment (vwxyzh) [inline src/Microsoft.DocAsCode.Build.Engine/Incrementals/PostProcessor/PostProcessInfo.cs]: readonly

**Pull request merged**